File: brute_gen.rb
#!/usr/bin/env ruby
require 'digest'  
def brute_gen(hash, hashtype, start_len, end_len)
    grn = "\e[32m"
    blu = "\e[34m"
    ylw = "\e[33m"
    red = "\e[31m"
    rst = "\e[0m"

    start_len = start_len.to_i
    end_len = end_len.to_i
         puts "\n#{ylw}Brute-Forcing...#{rst}"
         (start_len..end_len).each do |length|
            chars = "a" * length
            end_chars = "z" * length
            while chars <= end_chars

             hash_type = case hashtype
                 when "md5"
                     Digest::MD5.hexdigest(chars)
                 
                 when "sha1"
                     Digest::SHA1.hexdigest(chars)
             
                 when "sha256"
                     Digest::SHA256.hexdigest(chars)
             
                 when "sha384"
                     Digest::SHA384.hexdigest(chars)
             
                 when "sha512"
                     Digest::SHA512.hexdigest(chars)

                 else 
                    puts "\n#{red}Error: Invalid Hash Type!#{rst}"
                    puts "\n#{grn}Type#{rst} #{ylw}'help'#{rst} #{grn}for options.#{rst}"
                    return
               end

            if hash_type == hash
                puts "\n#{grn}Password Found:#{rst} #{ylw} #{chars} #{rst}"
                            return chars
            end
                   chars = chars.next
                              end
                            end
                  puts "\n#{red}Password Not Found!#{rst}"
end

File: core/cracker/brute_gen.go
package cracker

import "passcrax/core/utils"

import (
	"fmt"
)

const (
    borng = "\033[1;38;5;208m"
    bgrn = "\033[1;32m"
	bblu = "\033[1;34m"
	bred = "\033[1;31m"
	bylw = "\033[1;33m"
	grn = "\033[32m"
	blu = "\033[34m"
	ylw = "\033[33m"
	red = "\033[31m"
	orng = "\033[38;5;208m"
	rst = "\033[0m"
)

var charset = []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789.,;!?:'*#€@_&-+()/✓[]}{><∆§×÷=°^¢$¥£~|•√π`")

func BruteGen(targetHash string, hashtype string, startLen int, endLen int) string {
	if startLen < 1 || endLen < 1 || startLen > endLen {
		fmt.Printf("\n%s[!] Error: Invalid Length Parameters: Minimum length cannot be '0' !%s", bred, rst)
		return ""
	}

	fmt.Printf("\n%s[~] Brute-Forcing From Length %d To %d...%s\n", bylw, startLen, endLen, rst)

	for length := startLen; length <= endLen; length++ {
		fmt.Printf("%s[+] Trying Length: %d%s\n", bblu, length, rst)

		total := 1
		for i := 0; i < length; i++ {
			total *= len(charset)
		}

		for i := 0; i < total; i++ {
			text := make([]rune, length)
			n := i
			for j := length - 1; j >= 0; j-- {
				text[j] = charset[n%len(charset)]
				n /= len(charset)
			}

			word := string(text)
			hash_type, err := utils.HashFormats(word, hashtype)
			if err != nil {
				fmt.Printf("\n%s[!] Error: %s%s",
					red, err, rst)
				return ""
			}

			if hash_type == targetHash {
				fmt.Printf("\n%s[~] Password Found:%s %s%s%s\n", bgrn, rst, borng, word, rst)
				return word
			}
		}
	}

	fmt.Printf("\n%s[!] Password Not Found!%s\n", bred, rst)
	return ""
}
